certification/pyxis/pyxis.go

```
@@ -0,0 +1,179 @@
+}
+
+type pyxisEngine struct {
+	ApiToken  string
+	ProjectId string
+	Client    HTTPClient
+}
+
+func getPyxisUrl(path string) string {
+	return fmt.Sprintf("https://%s/%s/%s", viper.GetString("pyxis-host"), apiVersion, path)
+}
+
+func NewPyxisEngine(apiToken string, projectId string, httpClient HTTPClient) *pyxisEngine {
+	return &pyxisEngine{
+		ApiToken:  apiToken,
+		ProjectId: projectId,
+		Client:    httpClient,
+	}
+}
+
+func (p *pyxisEngine) GetRedHatLayers() (*[]Layer, error) {
+	return &[]Layer{}, nil
+}
+
+func (p *pyxisEngine) createImage(ctx context.Context, certImage *CertImage) (*CertImage, error) {
+	b, err := json.Marshal(certImage)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	req, _ := newRequestWithApiToken(ctx, http.MethodPost, getPyxisUrl("/images"), p.ApiToken)
+	req = withJson(req, b)
+
+	resp, err := p.Client.Do(req)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	var newCertImage CertImage
+	if err := json.Unmarshal(body, &newCertImage); err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	return &newCertImage, nil
+}
+
+func (p *pyxisEngine) createRPMManifest(ctx context.Context, imageId string, rpms []RPM) error {
+	rpmManifest := RPMManifest{
+		ImageID: imageId,
+		RPMS:    rpms,
+	}
+	b, err := json.Marshal(rpmManifest)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	req, _ := newRequestWithApiToken(ctx, http.MethodPost, getPyxisUrl(fmt.Sprintf("/images/id/%s/rpm-manifest", imageId)), p.ApiToken)
+	req = withJson(req, b)
+
+	_, err = p.Client.Do(req)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	return nil
+}
+
+func (p *pyxisEngine) getProject(ctx context.Context) (*CertProject, error) {
+	req, _ := newRequestWithApiToken(ctx, http.MethodGet, getPyxisUrl(fmt.Sprintf("/projects/certification/id/%s", p.ProjectId)), p.ApiToken)
```

are we going to ignore the error?

---

I'll add checks.